protocol/messenger_communities.go

```diff
@@ -1773,3 +1777,82 @@ func (m *Messenger) SyncCommunitySettings(ctx context.Context, settings *communi
 	chat.LastClockValue = clock
 	return m.saveChat(chat)
 }
+
+func (m *Messenger) ExtractDiscordDataFromImportFiles(filesToImport []string) (map[string]*discord_importer.DiscordCategory, []*discord_importer.DiscordExportedData, int, error) {
+
+	var discordChannels []*discord_importer.DiscordExportedData
+	discordCategories := map[string]*discord_importer.DiscordCategory{}
+	oldestMessageTime := 0
+
+	for _, fileToImport := range filesToImport {
+		filePath := strings.Replace(fileToImport, "file://", "", -1)
+		bytes, err := os.ReadFile(filePath)
+		if err != nil {
+			return nil, nil, 0, err
+		}
+
+		var discordExportedData discord_importer.DiscordExportedData
+
+		err = json.Unmarshal(bytes, &discordExportedData)
+		if err != nil {
+			return nil, nil, 0, err
+		}
+
+		discordExportedData.Channel.FilePath = filePath
+		categoryId := discordExportedData.Channel.CategoryId
+
+		discordCategory := discord_importer.DiscordCategory{
+			ID:   categoryId,
+			Name: discordExportedData.Channel.CategoryName,
+		}
+
+		_, ok := discordCategories[categoryId]
+		if !ok {
+			discordCategories[categoryId] = &discordCategory
+		}
+
+		discordChannels = append(discordChannels, &discordExportedData)
+
+		if len(discordExportedData.Messages) > 0 && (oldestMessageTime == 0 || int(discordExportedData.Messages[0].Timestamp) <= oldestMessageTime) {
+			// Exported discord channel data already comes with `messages` being
+			// sorted, starting with the oldest, so we can safely rely on the first
+			// message
+			oldestMessageTime = int(discordExportedData.Messages[0].Timestamp)
+		}
+	}
+	return discordCategories, discordChannels, oldestMessageTime, nil
```

Comment:
I've extracted this logic into a more generic function because we need it in the next iteration where we perform the actual import (which also receives the files to import from and needs to extract the data again as well [although it might just be a subset])